### ISSUE TITLE
About Vehicle

### ISSUE THREAD

**Issue by lo-th** [fabmax/physx-js-webidl#18]
Hello great port of 5.1 the speed up is amazing !
I see you don't expose vehicle function 
do you plan to add them that would be great.

**Comment by fabmax** [comment]
Yes I'm currently on that. The vehicle API has changed significantly compared to 4.1 so it's quite a bit of work...

**Comment by lo-th** [comment]
Awesome, i hope they have simplified  vehicle API because it was not easy to use

**Comment by lo-th** [comment]
Hi happy new year 
thank's for adding vehicle is great 
i have little missing and i don't find value in doc
vehicle.get_transmissionCommandState().set_targetGear(PxVehicleEngineDriveTransmissionCommandStateEnum.eAUTOMATIC_GEAR.value);
eAUTOMATIC_GEAR is not exposed, what the value ?

**Comment by fabmax** [comment]
Hmm, I think it is exposed...
Anyway, in the Nvidia docs you can click on the file name and then show source to see the actual values:
https://nvidia-omniverse.github.io/PhysX/physx/5.1.2/_build/physx/latest/program_listing_file_include_vehicle2_commands_PxVehicleCommandStates.h.html#program-listing-file-include-vehicle2-commands-pxvehiclecommandstates-h

For eAUTOMATIC_GEAR  it is 0xff

**Comment by lo-th** [comment]
ok thank's 
have to experiment now my car seem's go forward ? 
front wheel is 0 1 or 2 3 ? 

i wan't control car with keyboard
is with 
vehicle.getCommandState().setThrottle(1); // key for acceleration
vehicle.getCommandState().setNbBrakes(0);
vehicle.getCommandState().setSteer(0); // for turn

**Comment by lo-th** [comment]
ok i use get_wheelRigidBody1dStates(i).rotationAngle to update my wheel mesh
i guess Throttle value is also between 0.0 and 1.0
do you know how to reverse my direction key range is -1 1 

also do you know how change car shape to custom convex and get physics interaction with other object
i look you initialisation code in java version, wheel shape now work in js is great.

**Comment by fabmax** [comment]
For updating the wheels I would recommend using `vehicle.getBaseState().getWheelLocalPoses(i)` which also includes the suspension travel, steering etc.

**Comment by lo-th** [comment]
yes you right new version quarternion work

**Comment by lo-th** [comment]
they are something strange on your java setup ( is same as original physx example )
actorShapeLocalPose is ( 0, 0.83, 1.37 )
actorCMassLocalPose is ( 0, 0.55, 1.594 )
move center of mass to front bottom ok 
but why move z shape local pose ? that make model only over front wheel ?
